src/feature/nodelist/routerlist.c

```diff
@@ -546,6 +547,11 @@ router_add_running_nodes_to_smartlist(smartlist_t *sl, int need_uptime,
      * 0.3.1.0-alpha. */
     if (node_allows_single_hop_exits(node))
       continue;
+    /* Exclude relays that can not become a rendezvous for a hidden service
+     * version 3. */
+    if (rendezvous_v3 &&
```

Comment:
This refactoring looks okay, but is this tested?

Reply:
This commit is now in https://github.com/torproject/tor/pull/1888